homeassistant/components/esphome/dashboard.py

```diff
@@ -31,13 +32,13 @@ async def async_set_dashboard_info(
     """Set the dashboard info."""
     url = f"http://{host}:{port}"
 
-    # Do nothing if we already have this data.
-    if (
-        (cur_dashboard := hass.data.get(KEY_DASHBOARD))
-        and cur_dashboard.addon_slug == addon_slug
-        and cur_dashboard.url == url
-    ):
-        return
+    if cur_dashboard := async_get_dashboard(hass):
+        if cur_dashboard.addon_slug == addon_slug and cur_dashboard.url == url:
+            # Do nothing if we already have this data.
+            return
+        # Clear and make way for new dashboard
+        await cur_dashboard.async_shutdown()
+        del hass.data[KEY_DASHBOARD]
```

Comment:
I am clearing the dashboard in case the new one fails on `async_request_refresh`.

Another option would be to only shutdown/clear the old one after the new succeeds but I think this is cleaner.

Reply:
I also wonder if this should be backported to 2023.5.x, if dashboards are recreated often.
cc @balloob

Reply:
Dashboards are not recreated often, so not a really big deal.